### PR TITLE
UserConfiguration Properties + Profiles documentation

### PR DESCRIPTION
**Pull request by abujeda**
Updates to the ondemand.d/*.yml files page , adding the new UserConfiguration properties that can now be configured using these files.

As well, added basic information about profiles and how they work.

Added placeholders for the new documentation for:

Custom navbar
Custom help bar
Custom interactive apps menu
Custom pages

This is to be able to create a reference link from this page.

This PR is work in progress, I just wanted to start with the documentation for the new features that we have integrated and get early feedback on how to approach the documentation changes needed.

This is the page that this PR modifies:
https://osc.github.io/ood-documentation/latest/reference/files/ondemand-d-ymls.html

### CODE REVIEW

File: source/customizations/profiles.inc [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+      pinned_apps_group_by: "department"
+
+.. _profiles_selection_guide:
+
+Manual Profile Selection
+........................
+
+The selection of a profile can be done manually by the user or automatically based on request data.
+
+Links can be added the Help naviagtion menu to dinamically switch between profiles.
```
Comment (johrstrom) [inline source/customizations/profiles.inc]:
```suggestion
Links can be added the Help naviagtion menu to dynamically switch between profiles.
```

Comment (abujeda) [reply source/customizations/profiles.inc]: Fixed

File: source/customizations/profiles.inc [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+Links can be added the Help naviagtion menu to dynamically switch between profiles.
+These links can be configured using the configuration property ``link_profiles``.
+
+.. code-block:: yaml
+
+  profile_links:
+    - id: ""
+      name: "Root Profile"
+      icon: "cog"
```
Comment (johrstrom) [inline source/customizations/profiles.inc]: Isn't this `icon_uri`?

Comment (johrstrom) [reply source/customizations/profiles.inc]: No it is `icon`, we use `icon_uri` in the other places...

File: source/reference/files/ondemand-d-ymls.rst [hunk 3]
```diff
@@ -3,64 +3,127 @@
...
 
-    An array of pinned app objects specifying what apps to pin to the dashboard.
-    See the :ref:`documentation on pinned apps <dashboard_pinned_apps>` for details
-    and examples.
+.. _profile_properties:
 
+Configuration Properties with profile support
+---------------------------------------------
+
+.. describe:: dashboard_header_img_logo: (String, null)
```
Comment (johrstrom) [inline source/reference/files/ondemand-d-ymls.rst]:
```suggestion
.. describe:: dashboard_header_img_logo (String, null)
```

I don't think you did this. Indeed, I think I did when I started this file, but I'm just noticing it now that the other pages don't use a colon here.  Let's remove it just for consistency.

File: source/customizations/profiles.inc (hunk 2 above)
Comment (abujeda) [reply source/customizations/profiles.inc]: I notice this while doing the documentation.
I am working on a small PR to make this consistent with the `NavBar` configuration.
After the change, the config above will look like this:

```yaml
  profile_links:
    - title: "Root Profile"
      profile: ""
      icon_uri: "fas://cog"
    - title: "Rstudio Profile"
      profile: rstudio_group
      icon_uri: "fas://user"
    - title: "Jupiter Profile"
      profile: jupyter_group
      icon_uri: "fas://user"
```
Will update the documentation when the PR is merged

File: source/reference/files/ondemand-d-ymls.rst (hunk 3 above)
Comment (abujeda) [reply source/reference/files/ondemand-d-ymls.rst]: Fixed

File: source/customizations/profiles.inc (hunk 2 above)
Comment (johrstrom) [reply source/customizations/profiles.inc]: I am wondering if we should just use `icon` everwhere and force `fas`. Or at least interpret the string and see if it's a full URI. if so, we do nothing, if not, then we default to `fas`.

Comment (abujeda) [reply source/customizations/profiles.inc]: A quick scan through OOD, `icon` is used in the apps `manifest`, `icon_uri` is used in the form for`batch_connect` apps and in new navigation configuration.

Everywhere is all of OOD or just all things related to the new navigation?
If just for the new navigation, I can apply the changes to the PR to update the configuration for the profile links

Comment (johrstrom) [reply source/customizations/profiles.inc]: everywhere that's safe to do it.  We'd have to consider compatibility when modifying `manifest` and `batch_connect` forms, so we can defer that or tackle it as a separate thing entirely.

In any case, I would +1 we launch the new navigation with just `icon` that can infer scheme and default to fas.

Comment (abujeda) [reply source/customizations/profiles.inc]: OK - will implement the change for the new navigation.

**Comment by abujeda** [comment]
Regarding documentation, I have more changes for the other new features that we have integrated.
Is it best to create a single PR with all the changes and review or create smaller PRs grouped by feature?

**Comment by abujeda** [comment]
Closing, superseded by #760

**Pull request closed without merging**